<launch>
	<group ns="fmSensors">
		<!-- GPS -->
		<!--node pkg="serial_string" name="gps_serial" type="serial_string_node" output="screen"  respawn="true">
			<param name="publisher_topic" value="/fmData/gps_rx" />	 
			<param name="subscriber_topic" value="/fmData/gps_tx" />
			<param name="device" value="/dev/ttyUSB1" />	
			<param name="baudrate" value="115200" />
			<param name="termination_character" value="10"/>
		</node-->
		
	    <node pkg="serial_nmea" name="nmea_converter" type="nmea_node" output="screen">
			<param name="str_to_msg_sub" value="/fmData/gps_rx" />   
			<param name="str_to_msg_pub" value="/fmData/nmea_from_gps" />
			<param name="msg_to_str_sub" value="/fmData/nmea_to_gps" />   
			<param name="msg_to_str_pub" value="/fmData/gps_tx" />           
			<param name="use_nmea_checksum" type="boolean" value="true" />
	    </node>

		<!-- IMU -->
		<node pkg="serial_string" name="imu_serial" type="serial_string_node" output="screen" respawn="true">
			<param name="publisher_topic" value="/fmData/imu_rx" /> 
			<param name="subscriber_topic" value="/fmData/imu_tx" />
			<param name="device" value="/dev/ttyUSB2" />	
			<param name="baudrate" value="115200" />
			<param name="termination_character" value="10"/>
		</node> 
		
		<node pkg="vectornav_vn100" name="imu_converter"  type="vectornav_vn100_node" output="screen" >
			<param name="vectornav_vn100_sub" value="/fmData/imu_rx"/>
			<param name="imu_pub" value="/fmInformation/imu"/>
			<param name="frame_id" value="imu_link"/>
			<param name="use_enu" value="true"/>
			<param name="covariance_x" value="0.001"/>
			<param name="covariance_y" value="0.001"/>
			<param name="covariance_z" value="0.001"/>
		</node>	
	</group>
	
	<group ns="fmProcessors">
		<node pkg="differential_odometry" name="differential_odometry" type="differential_odometry_node" output="screen" >
			<param name="enc_left_sub" value="/fmInformation/enc_left" />
			<param name="enc_right_sub" value="/fmInformation/enc_right" />
			<param name="imu_sub" value="/fmInformation/imu" />
			<param name="odom_pub" value="/fmKnowledge/encoder_odom" />
			<param name="yaw_angle_source" value="imu_angular_velocity"/>  
			<param name="imu_angular_velocity_yaw_axis" value="-d" /> 
			<param name="vehicle_frame" value="base_footprint" />      
			<param name="odom_estimate_frame" value="/wheel_odom" />      
		</node>
		
		<!--node pkg="utm_to_odom" name="gps_odometry" type="utm_to_odom_node" output="screen"> 
	    	<param name="subscribe_topic_id" value="/fmInformation/gpgga" />
			<param name="publish_topic_id" value="/fmKnowledge/gps_odom" />
			<param name="odom_frame_id" value="odom_combined" />
			<param name="child_odom_frame_id" value="gps_link" />
			<param name="receive_n_before_publish" value="10"/>
			<param name="gps_variance" value="1"/>
		</node-->
	
	</group>
</launch>